test/nodes/test_connector.py

```diff
@@ -119,3 +58,79 @@ def test_crawler_return_document(tmp_path):
             file_content = json.load(read_file)
             assert file_content["meta"] == document.meta
             assert file_content["content"] == document.content
+
+
+#
+# Unit tests
+#
+
+
+def test_crawler_url_none_exception(tmp_path):
+    crawler = Crawler(tmp_path)
+    with pytest.raises(ValueError):
+        crawler.crawl()
+
+
+def test_crawler_depth_0_single_url(test_url, tmp_path):
+    crawler = Crawler(output_dir=tmp_path)
+    paths = crawler.crawl(urls=[test_url + "/index.html"], crawler_depth=0)
+    assert len(paths) == 1
+    assert content_match(crawler=crawler, base_url=test_url, page_name="index", crawled_page=paths[0])
```

Comment:
Cosmetics: I think var names are verbose enough, we can avoid passing key args and make the function call more readable. I would also change the signature of `content_match`, the caller has the `url` already and we can spare a parameter, like:
```python
def test_crawler_depth_0_single_url(test_url, tmp_path):
    crawler = Crawler(output_dir=tmp_path)
    url = test_url + "/index.html"
    paths = crawler.crawl(urls=[url], crawler_depth=0)
    assert len(paths) == 1
    assert content_match(crawler, url, paths[0])
```